app/windows/msi/build.js

```diff
@@ -0,0 +1,43 @@
+// You need to install wix tool set first
+// Reference https://ourcodeworld.com/articles/read/927/how-to-create-a-msi-installer-in-windows-for-an-electron-framework-application
+const { MSICreator } = require('electron-wix-msi');
+const path = require('path');
+const info = require('../../package.json');
+
+const APP_DIR = path.resolve(__dirname, '../../dist/win-unpacked');
+const OUT_DIR = path.resolve(__dirname, '../../dist');
+
+const nameOptions = {
+  productName: info.productName,
+  version: info.version,
+  os: 'win',
+  arch: 'x86',
+};
+
+// Generate the exe name from electron-builder's artifactName
+let exeName = info.build.artifactName.split('.')[0];
+Object.entries(nameOptions).forEach(([key, value]) => {
+  exeName = exeName.replace(`\${${key}}`, value);
+});
+
+const msiOptions = {
```

Comment:
I reviewed the configuration options which are documented here: https://github.com/felixrieseberg/electron-wix-msi#configuration

> shortName (optional, string) - A short name for the app, used wherever spaces and special characters are not allowed. Will use the name if left undefined.

I wonder if we need a shortName here? It's a place for where spaces and special characters are not allowed. For "Headlamp" it is ok to not have one, but for names like "Sxxxx Bxxxx" it's not.

> upgradeCode (string, optional) - A unique UUID used by your app to identify itself. This module will generate one for you, but it is important to reuse it to enable conflict-free upgrades.

This `upgradeCode` seems important to keep track of and check.

> arch (string, optional) - Defines the architecture the MSI is build for. Values can be either x86 or x64. Default's to x86 if left undefined.

How is arch handled here? We need a x64 only? Seems we are setting 'x86'.

Perhaps we could override some of the images shown in the installer?https://github.com/felixrieseberg/electron-wix-msi#ui-configuration-optional

Reply:
All good points.
I will address all but the images, which we can add later.